netket/optimizer/linear_operator.py

```diff
@@ -67,15 +47,30 @@ class AbstractSMatrix:
 
     """
 
-    sr: SR
-    """Parameters for the solution of the system."""
+    diag_shift: float = 0.00
+    """Diagonal shift added to the S matrix."""
 
     # PUBLIC API: METHOD TO EXTEND IF YOU WANT TO DEFINE A NEW S object
     def __matmul__(self, vec):
         raise NotImplementedError()
 
-    # PUBLIC API: METHOD TO EXTEND IF YOU WANT TO DEFINE A NEW S object
-    def solve(self, y: PyTree, **kwargs) -> PyTree:
+    def __add__(self, eps):
+        return self.replace(diag_shift=self.diag_shift + eps)
+
+    # PUBLIC API: METHOD TO EXTEND Optionally IF YOU WANT TO DEFINE A NEW S object with
+    # custom logic
+    def _solve(self, solve_fun, y: PyTree, *, x0: Optional[PyTree], **kwargs) -> PyTree:
+        # dont pass x0 if it's unset.
+        # some solvers might not need/require it.
+        if x0 is not None:
+            kwargs["x0"] = x0
+
+        raise solve_fun(self, y, **kwargs)
```

Comment:
`raise`?

Reply:
? There was a typo. But I don't see why raising

Reply:
I mean, shouldn't that be `return`?

Reply:
indeed. Sorry, Brain fart, I was looking at x0 and didn't look at the raise.